pkg/service/providerindex/legacy.go

```diff
@@ -89,6 +98,10 @@ func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Mult
 		}
 
 		results = append(results, pr)
+
+		if err := ls.claimsCache.Set(ctx, claimCid, claim, true); err != nil {
```

Comment:
Won't it stay in the cache forever if we always store after retrieve?

Reply:
That sounds reasonable...but the point of caching it is to ensure it's super fast on subsequent reads. Here we'd be enforcing a write on read for every request, at some number of requests it's going to be slower on average than letting it fall out of the cache and revalidate. Also if it is removed from the underlying store, and keeps getting refreshed then it will never expire.

Reply:
thanks! That reasoning makes total sense. I updated the code so that claims are not written to the cache if they were fetched from there.